Rewrite defense mechanisms for Cache Probing (#54)
xsleaks/wiki: content/docs/attacks/cache-probing.md

- Request headers such as Content-Type, Accept, Accept-Language, etc that may cause the server to fail (more application dependent).
- Other request properties.

Often some of these methods might be considered a bug in the browser (e.g. [this bug](https://bugs.chromium.org/p/chromium/issues/detail?id=959789#c9)).

## Defense

> Currently there exist no reliable defense mechanisms that would allow websites to protect against Cache Probing attacks. Nonetheless, a website could try to mitigate the attack surface by deploying [Cache Protections]({{< ref "cache-protections.md" >}}), such as:

Reviewer: Is it really true that there are no reliable defenses? As I understand it, cache-control and random tokens both are a pretty reliable defense mechanism?

```suggestion
Currently there are no default protections against cache probing attacks that applications can rely on. Nonetheless, a website mitigate the attack by deploying [Cache Protections]({{< ref "cache-protections.md" >}}), such as:
```

Author: That's a tricky question. Even when we split the cache it's still not fixing the cache leaks completely. You can still probe the cache via counting the number of network requests (socket exhaustion) for instance, probably other ways too. And you can always purge something from cache by exceeding the available disk space for the cache.. 

```suggestion
Currently there are no good defense mechanisms that would allow websites to fully protect against Cache Probing attacks. Nonetheless, a website could might mitigate the attack surface by deploying [Cache Protections]({{< ref "cache-protections.md" >}}), such as:
```

Let me know what you think about that phrasing.

Author: I am merging the above section, we can add an additional PR to fix that one sentence.

> Currently there exist no reliable defense mechanism that would allow websites to protect against Cache Probing attacks. Nonetheless, a website could try to mitigate the attack surface by deploying defenses in depth, such as:

Author: Forgot about it, thanks. That way will be much cleaner. Applied the suggestion in https://github.com/xsleaks/wiki/pull/54/commits/5d9a618ecb615ec067d057bd213aa76b3c51cedb

- [Vary: Sec-Fetch-Site]({{< ref "../defenses/opt-in/fetch-metadata.md#fetch-metadata--cache-probing-attacks" >}}) allows applications to force cache segregation by a group of origins.
- [Subresource Protections]({{< ref "../defenses/design-protections/subresource-protections.md" >}}) allow application to set random tokens in URLs to make them unpredictable from attackers. Useful for both authenticated and unauthenticated subresources.
- [`Cache-control: no-store`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) signals the browser to not cache the vulnerable resource.

> A promising defense against the attack is [Partitioned HTTP Cache]({{< ref "../defenses/secure-defaults/partitioned-cache.md" >}}) feature implemented in browsers. This protection prevents an attacker's origin from interfering with cached resources of other origins.

Reviewer: Just trying to make this bit read more like a sentence. 

```suggestion
A promising defense against this attack is [partitioning the HTTP cache]({{< ref "../defenses/secure-defaults/partitioned-cache.md" >}}) by the requesting origin. This browser provided protection prevents an attacker's origin from interfering with cached resources of other origins.
```

Reviewer: Or alternatively if we wanted to keep "Partitioned HTTP Cache" as a key phrase (either of these is fine with me):

```suggestion
[Partitioned HTTP Caches]({{< ref "../defenses/secure-defaults/partitioned-cache.md" >}}) are a promising new browser defense mechanism that splits the cache by the requesting origin. This protection prevents an attacker's origin from interfering with cached resources of other origins.
```
